com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/protocol/AbstractProtocolServer.java

```diff
@@ -131,7 +197,11 @@ private void processData() {
                             logger.log(Level.SEVERE, String.format("Dispatch debug protocol error: %s", e.toString()), e);
                         }
                     } else if (message.type.equals("response")) {
-                        // handle response.
+                        Messages.Response response = JsonUtils.fromJson(messageData, Messages.Response.class);
+                        Consumer<Messages.Response> cb = fetchPendingRequestCallback(response.request_seq);
```

Comment:
This is not systematic. We need to expose an observable of responses from vs code and let others subscribe to it.